pkg/cloudprovider/types.go

```diff
@@ -30,6 +30,8 @@ import (
 	"github.com/aws/karpenter-core/pkg/utils/resources"
 )
 
+type DriftReason string
```

Comment:
It probably makes more sense to define the Drift Reason in the cloud provider neutral code, then have the IsMachineDrifted just refer to that enum type, since there are some cloud provider agnostic Drift Reasons. Thoughts?

Reply:
The only overlap we will have would Static drift. Outside of that I don't see any other cases for agnostic reasons.